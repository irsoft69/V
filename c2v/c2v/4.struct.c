#include <stdio.h>

struct User {
	char *name;
	int age;
};

// lots of structs are typedef'ed in C
typedef struct {
	char *name;
	int age;
} TUser;

/*
enum Color {
	red, green, blue,
};
*/

void handle_user(struct User user) {

}

void handle_tuser(TUser user) {

}

int main() {
	struct User user;
	user.age = 20;
	user.name = "Bob";
	printf("age=%d", user.age);
	handle_user(user);
	// TODO
	// struct User user2 = { .age = 30, .name = "Peter" };
	TUser user2;
	user2.age = 30;
	user2.name = "Peter";
	handle_tuser(user2);
	return 0;
}
