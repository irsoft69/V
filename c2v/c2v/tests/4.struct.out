struct User {
	name &char
	age  int
}

struct TUser {
	name &char
	age  int
}

fn handle_user(user User) {
}

fn handle_tuser(user TUser) {
}

fn main() {
	user := User{} // struct init "User" struct User:struct User
	user.age = 20
	user.name = 'Bob'
	printf('age=%d', user.age)
	handle_user(user)
	user2 := TUser{} // struct init "TUser" TUser:TUser
	user2.age = 30
	user2.name = 'Peter'
	handle_tuser(user2)
}
